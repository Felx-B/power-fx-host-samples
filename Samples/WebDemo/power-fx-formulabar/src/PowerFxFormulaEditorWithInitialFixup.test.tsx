/*!
 * Copyright (C) Microsoft Corporation. All rights reserved.
 */

import 'jest';

import * as Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { PowerFxFormulaEditorWithInitialFixup } from './PowerFxFormulaEditorWithInitialFixup';
import { IDisposable } from './PowerFxFormulaEditor.types';

const messageProcessor = {
  addListener: (listener: (data: string) => void): IDisposable => {
    return {
      dispose: () => null
    };
  },
  sendAsync: async (data: string): Promise<void> => {
    // do nothing
  }
};

it('basic test', () => {
  const element = Enzyme.shallow(
    <PowerFxFormulaEditorWithInitialFixup
      getDocumentUriAsync={async () => 'powerfx://demo?a=1&b=2'}
      defaultValue={''}
      messageProcessor={messageProcessor}
      maxLineCount={1}
      minLineCount={1}
      editorFocusOnMount={true}
    />
  );
  expect(toJson(element)).toMatchSnapshot();
});

it('with monacoEditorOptions test', () => {
  const element = Enzyme.shallow(
    <PowerFxFormulaEditorWithInitialFixup
      getDocumentUriAsync={async () => 'powerfx://demo?a=1&b=2'}
      defaultValue={''}
      monacoEditorOptions={{
        fontSize: 35,
        fixedOverflowWidgets: false
      }}
      messageProcessor={messageProcessor}
      maxLineCount={1}
      minLineCount={1}
      editorFocusOnMount={true}
    />
  );
  expect(toJson(element)).toMatchSnapshot();
});
