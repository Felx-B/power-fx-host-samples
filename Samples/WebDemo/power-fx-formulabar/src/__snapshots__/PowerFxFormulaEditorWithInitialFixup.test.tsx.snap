// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test 1`] = `
<StyledPowerFxFormulaEditorBase
  defaultValue=""
  editorFocusOnMount={true}
  getDocumentUriAsync={[Function]}
  maxLineCount={1}
  messageProcessor={
    Object {
      "addListener": [Function],
      "sendAsync": [Function],
    }
  }
  minLineCount={1}
/>
`;

exports[`with monacoEditorOptions test 1`] = `
<StyledPowerFxFormulaEditorBase
  defaultValue=""
  editorFocusOnMount={true}
  getDocumentUriAsync={[Function]}
  maxLineCount={1}
  messageProcessor={
    Object {
      "addListener": [Function],
      "sendAsync": [Function],
    }
  }
  minLineCount={1}
  monacoEditorOptions={
    Object {
      "fixedOverflowWidgets": false,
      "fontSize": 35,
    }
  }
/>
`;
